src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4encoder/HaivisionX4EncoderCommunicator.java

```diff
@@ -803,21 +944,787 @@ else if (EncodingProfile.MAIN_10.getName().equals(encodingProfileInput) || Encod
+				addAdvanceControlProperties(advancedControllableProperties, bandwidthControlProperty);
+				outputResponseItem.setBandwidthOverhead(value);
+				break;
+			case PARAMETER_MTU:
+				if (Integer.parseInt(value) < HaivisionConstant.MIN_OF_MTU) {
+					value = Integer.toString(HaivisionConstant.MIN_OF_MTU);
+				}
+				if (Integer.parseInt(value) > HaivisionConstant.MAX_OF_MTU) {
+					value = Integer.toString(HaivisionConstant.MAX_OF_MTU);
+				}
+
+				AdvancedControllableProperty mtuControlProperty = controlNumeric(extendedStatistics, property, value);
+				addAdvanceControlProperties(advancedControllableProperties, mtuControlProperty);
+				outputResponseItem.setMtu(value);
+				break;
+			case PARAMETER_TTL:
+				if (Integer.parseInt(value) < HaivisionConstant.MIN_OF_TTL) {
+					value = Integer.toString(HaivisionConstant.MIN_OF_TTL);
+				}
+				if (Integer.parseInt(value) > HaivisionConstant.MAX_OF_TTL) {
+					value = Integer.toString(HaivisionConstant.MAX_OF_TTL);
+				}
+
+				AdvancedControllableProperty ttlControlProperty = controlNumeric(extendedStatistics, property, value);
+				addAdvanceControlProperties(advancedControllableProperties, ttlControlProperty);
+				outputResponseItem.setTtl(value);
+				break;
+			case PARAMETER_TOS:
+				//Support hex value only
+				if (value.startsWith("0x")) {
+					String valueCopy = value.replace("0x", "");
+					try {
+						int decTos = Integer.parseInt(valueCopy, 16);
+						int decMaxTos = Integer.parseInt(HaivisionConstant.MAX_OF_TOS, 16);
+						int decMinTos = Integer.parseInt(HaivisionConstant.MIN_OF_TOS, 16);
+						if (decTos < decMinTos) {
+							value = "Ox" + HaivisionConstant.MIN_OF_TOS;
+						}
+						if (decTos > decMaxTos) {
+							value = "0x" + HaivisionConstant.MAX_OF_TOS;
+						}
+
+						AdvancedControllableProperty toSControlProperty = controlText(extendedStatistics, property, value);
+						addAdvanceControlProperties(advancedControllableProperties, toSControlProperty);
+						outputResponseItem.setTos(value);
+					} catch (Exception e) {
+						throw new NumberFormatException("Value of ParameterToS is invalid. TOS must be between 0 to 255");
+					}
+				}
+				break;
+			case SOURCE_ADD_AUDIO:
+				Map<String, Audio> addSourceAudio = sourceAudioResponse.get(streamName);
+				editSourceAudioCreateOutputStream(streamName, addSourceAudio, extendedStatistics, advancedControllableProperties);
+				break;
+			case APPLY_CHANGE:
+				OutputResponse outputResponse = convertOutputStreamByValue(extendedStatistics, streamName);
+
+				// sent request to apply all change for all metric
+				setOutputStreamApplyChange(outputResponse.payLoad(), outputResponse.getId());
+
+				//sent request to action for the metric
+				setActionOutputStreamControl(streamName, outputResponse);
+				isEmergencyDelivery = false;
+				break;
+			case CANCEL:
+				isEmergencyDelivery = false;
+				break;
+			default:
```

Comment:
also log entry wont hurt here :) Please check other switch-cases for this

Reply:
I fixed it